src/node/server.ts

```diff
@@ -189,6 +215,11 @@ export abstract class Server {
 		request: http.IncomingMessage,
 	): Promise<Response>;
 
+	protected abstract handleSSH(
```

Comment:
I think some of this could be split out, but I think they can't exist completely separately because the `SSHServer` is accessed over HTTP, so the `HTTPServer` would need a reference to it to defer to when `/ssh` is hit, making it more of a parent / child relationship.

I'm down to split things out in any way, but I guess I don't fully understand the purpose of it being this way as-is since there's only one implementation of `Server` right now.

Reply:
Mmm yup that's why we'd need to pass the ssh server to the http server (or at least the port number---it could even be a generic proxy method that proxies the endpoints to ports that you give it). It doesn't feel like we necessarily need a tight coupling between the http and ssh servers is what I'm driving at, I guess.

My reasoning for advocating a split is to make the classes more focused (like how a method should do one thing) which I think should make it easier to reason about the code as well as increase composability in case for some reason we want to spawn a second HTTP server in the future or if we wanted to make the SSH server implementation publicly available or something like that.

For example, suppose we decided the webview should be a hosted on a separate port again for security/isolation reasons. We probably wouldn't want to just extend `Server` since we'd be inheriting an entire SSH server along with the HTTP server so our options would be to split it up in a composable way or to manage a second HTTP server in `Server` which would be a bit tricky. 

I don't wanna push for composability just for the sake of being composable so if this seems like overkill it's all good to me~ :cake:

Reply:
Oh as a practical example of some other potential benefits, if the http and ssh servers were decoupled in this way then you wouldn't need any `this.sshServer` guards or `!`. Basically you reduce the number of special edge cases, I think.

Reply:
I've been trying to keep the LOC changed as low as possible since I don't live here and didn't want to start re-arranging the furniture, but if you'd like me to take a crack at breaking it out, I'd be happy to give it a go.

Otherwise, that might be a good thing to do later if we ever split out the SSH server into its own module as you suggested (Which I think is a great idea, I think this has potential to be a very cool programmable ssh / sftp server module!)

Reply:
sgtm